overlord/devicestate/systems.go

```diff
@@ -83,28 +83,39 @@ func systemFromSeed(label string) (*System, error) {
 	if err != nil {
 		return nil, fmt.Errorf("cannot obtain brand: %v", err)
 	}
-	system := System{
+	system := &System{
 		Current: false,
 		Label:   label,
 		Model:   model,
 		Brand:   brand,
 		Actions: defaultSystemActions,
 	}
-	return &system, nil
+	if currentSystem != nil && isCurrentSystem(currentSystem, system) {
+		system.Current = true
+		system.Actions = currentActions
+	}
+	return system, nil
 }
 
-func currentSystemForMode(st *state.State, mode string) (*seededSystem, error) {
+func currentSystemForMode(st *state.State, mode string) (currentSys *seededSystem, actions []SystemAction, err error) {
```

Comment:
wondering if it's worth having a struct for the result:

```
type currentSystem struct {
    cur *seededSystem
    actions []SystemAction
}
```

it could also carry with an appropriate name the isCurrentSystem helper, maybe as just isCurrent(*System)  ?

if it's defined on (*currentSystem) it could work on nil as well returning always false.

Reply:
Nice idea. I actually embedded the seededSystem.